fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/KeyExpressionExpansionVisitor.java

```diff
@@ -294,31 +297,18 @@ public GraphExpansion visitExpression(@Nonnull final NestingKeyExpression nestin
                     final SelectExpression selectExpression =
                             sealedBaseAndChildExpansion.buildSelect();
                     final Quantifier childQuantifier = Quantifier.forEach(Reference.initialOf(selectExpression));
-
-                    final var childExpansionValues =
-                            childExpansion.getResultColumns()
-                                    .stream()
-                                    .map(Column::getValue)
-                                    .collect(ImmutableList.toImmutableList());
                     final var childResultValue = selectExpression.getResultValue();
-                    final var pulledUpValuesMap =
-                            childResultValue.pullUp(childExpansionValues, EvaluationContext.empty(),
-                                    AliasMap.emptyMap(), ImmutableSet.of(), childQuantifier.getAlias());
-                    final ImmutableList<Column<? extends Value>> pulledUpExpansionColumns =
-                            childExpansionValues.stream()
-                                    .map(value -> {
-                                        if (!pulledUpValuesMap.containsKey(value)) {
-                                            throw new RecordCoreException("could not pull expansion value " + value)
-                                                    .addLogInfo(LogMessageKeys.VALUE, value);
-                                        }
-                                        return pulledUpValuesMap.get(value);
-                                    })
-                                    .map(Column::unnamedOf)
-                                    .collect(ImmutableList.toImmutableList());
 
-                    return sealedBaseAndChildExpansion
-                            .builderWithInheritedPlaceholders()
+                    final var pulledUpExpansionColumns =
+                            pullUpResultColumns(childExpansion, childResultValue, childQuantifier);
+
+                    final var pulledUpPlaceholders =
+                            pullUpPlaceholders(childExpansion, childResultValue, childQuantifier);
+
+                    return GraphExpansion.builder()
                             .addQuantifier(childQuantifier)
+                            .addAllPredicates(pulledUpPlaceholders)
+                            .addAllPlaceholders(pulledUpPlaceholders)
```

Comment:
Could you speak to what the "problem" was here? Like, is the issue that without pulling up the placeholders on nested fields, matching logic down the line wasn't able to associate sargable predicates with the index, or something like that?

It does sort of concern me that there are no test cases that are added here that depend on the new behavior. There are some tests of nested repeated (aggregate) queries in `FDBNestedRepeatedQueryTest`. Those could be modified to include some more cases, or we could add some more YAML tests with some nested repeated queries

Reply:
Okay, yeah, the actual difference here makes sense. It may be useful to note the reason in a code comment

> This problem only occurs in aggregate indexes over a fanned out group or aggregation column. Thus, things like `FDBNestedRepeatedQueryTest` are not able to pick up a difference in behavior.

That test class is designed, amongst other things, to test aggregate indexes over fanned out groups or values. I think none of the problematic cases that exhibit those differences are currently under test, though, which is why none of them failed. For example, there may not be queries that put predicates on the fanned-out columns of the select-where, and so they don't require the duplicated placeholders.